File: year2/advanced_programming/Lab7/lab7/Repository/Repository.cs
using lab7.Domain;
using System;
using System.Collections.Generic;
using System.Text;

namespace lab7.Repository
{
    interface Repository<ID, E> where E : Entity<ID>
    {
        E FindOne(ID id);
        IEnumerable<E> FindAll();
        E Save(E entity);
        E Delete(ID id);
        E Update(E entity);
    }
}

File: year2/systems-design-implementation/Lab3/Lab3/src/main/java/repo/interfaces/Repository.java
package repo.interfaces;

import utils.Entity;

import java.sql.SQLException;
import java.util.Optional;

public interface Repository<ID, E extends Entity<ID>> {

    Optional<E> findOne(ID id) throws SQLException;

    Iterable<E> findAll() throws SQLException;

    Optional<E> add(E entity) throws SQLException;

    Optional<E> delete(ID id) throws SQLException;

    Optional<E> update(E entity) throws SQLException;

}
